python/pyarrow/_s3fs.pyx

```diff
@@ -236,16 +236,24 @@ cdef class S3FileSystem(FileSystem):
             S3FileSystem(proxy_options={'scheme': 'http', 'host': 'localhost',
                                         'port': 8020, 'username': 'username',
                                         'password': 'password'})
-    allow_bucket_creation : bool, default False
+    allow_bucket_creation : boolean, default False
         Whether to allow CreateDir at the bucket-level. This option may also be
         passed in a URI query parameter.
-    allow_bucket_deletion : bool, default False
+    allow_bucket_deletion : boolean, default False
         Whether to allow DeleteDir at the bucket-level. This option may also be
         passed in a URI query parameter.
+    check_directory_existence_before_creation: boolean, default false
+        Whether to allow pessimistic directory creation in CreateDir function
```

Comment:
It's rather unclear what "pessimistic" means in this context (but I see this term is used on the C++ side as well). Also, we shouldn't mention "CreateDir" as that is a C++ function the python user is not familiar with. We can use a generic term, something like "when creating a directory"